src/libs/actions/Task.js

```diff
@@ -0,0 +1,219 @@
+}
+
+/**
+ * Assign a task to a user
+ * Function title is createTask for consistency with the rest of the actions
+ * and also because we can create a task without assigning it to anyone
+ * @param {String} currentUserEmail
+ * @param {String} parentReportID
+ * @param {String} title
+ * @param {String} description
+ * @param {String} assignee
+ *
+ */
+
+function createTaskAndNavigate(currentUserEmail, parentReportID, title, description, assignee = '') {
+    // Create the task report
+    const optimisticTaskReport = ReportUtils.buildOptimisticTaskReport(currentUserEmail, assignee, parentReportID, title, description);
+
+    // Grab the assigneeChatReportID if there is an assignee and if it's not the same as the parentReportID
+    // then we create an optimistic add comment report action on the assignee's chat to notify them of the task
+    const assigneeChatReportID = lodashGet(ReportUtils.getChatByParticipants([assignee]), 'reportID');
+    let optimisticAssigneeAddComment;
+    if (assigneeChatReportID && assigneeChatReportID !== parentReportID) {
+        optimisticAssigneeAddComment = ReportUtils.buildOptimisticAddCommentReportAction(
+            `${currentUserEmail} has[created a task for you](tbd/r/${optimisticTaskReport.reportID}): ${title}`,
+        );
+        optimisticAssigneeAddComment.reportAction.message[0].taskReportID = optimisticTaskReport.reportID;
+    }
+
+    // Create the CreatedReportAction on the task
+    const optimisticTaskCreatedAction = ReportUtils.buildOptimisticCreatedReportAction(optimisticTaskReport.reportID);
+
+    const optimisticAddCommentReport = ReportUtils.buildOptimisticAddCommentReportAction(
+        `[Created a task](tbd/r/${optimisticTaskReport.reportID}): ${title}`,
+    );
+    optimisticAddCommentReport.reportAction.message[0].taskReportID = optimisticTaskReport.reportID;
+
+    const optimisticData = [
+        {
+            onyxMethod: Onyx.METHOD.SET,
+            key: `${ONYXKEYS.COLLECTION.REPORT}${optimisticTaskReport.reportID}`,
+            value: optimisticTaskReport,
+        },
+        {
+            onyxMethod: Onyx.METHOD.SET,
+            key: `${ONYXKEYS.COLLECTION.REPORT_ACTIONS}${optimisticTaskReport.reportID}`,
+            value: {[optimisticTaskCreatedAction.reportActionID]: optimisticTaskCreatedAction},
+        },
+        {
+            onyxMethod: Onyx.METHOD.MERGE,
+            key: `${ONYXKEYS.COLLECTION.REPORT_ACTIONS}${parentReportID}`,
+            value: {[optimisticAddCommentReport.reportAction.reportActionID]: optimisticAddCommentReport.reportAction},
+        },
+    ];
+
+    if (optimisticAssigneeAddComment) {
+        optimisticData.push({
+            onyxMethod: Onyx.METHOD.MERGE,
+            key: `${ONYXKEYS.COLLECTION.REPORT_ACTIONS}${assigneeChatReportID}`,
+            value: {[optimisticAssigneeAddComment.reportAction.reportActionID]: optimisticAssigneeAddComment.reportAction},
+        });
+    }
+
+    const successData = [];
```

Comment:
This caused a regression. The `successData` array is empty but we should be clearing the pendingActions. (Coming from https://github.com/Expensify/App/issues/22907)